### PR TITLE
Add parameter substitution to `download.sh`

### PR DESCRIPTION
**Pull request by christopherwoodall**
Utilize parameter substitution in `download.sh` to allow both `PRESIGNED_URL` and `TARGET_FOLDER` to be assigned via environmental variables. 

This removes the need to manually edit/sed the file once a developer receives the confirmation url.

### CODE REVIEW

**Comment by ckxng** [comment]
picking this up, I am reviewing several of the outstanding PRs for download.sh

**Comment by ckxng** [comment]
@TheFinalJoke take a look at this

**Comment by ckxng** [comment]
The user-friendliness of prompting for the signed URL is nice, but I also see the value in accepting ENV vars.  I would be interested to see a diff that preserves both behaviors.  (Prompt if ENV var not set?)

**Comment by ckxng** [comment]
Very good, I think this one should merge before #441, so that this behavior is in place before [krrishnarraj](https://github.com/krrishnarraj) syncs with main.

**Review by samuelselvan — APPROVED** [review]
This looks good to me. @ruanslv any thoughts?

File: README.md [hunk 1]
```diff
@@ -14,6 +14,14 @@ See [UPDATES.md](UPDATES.md).
...
 ⚠️ **7/18: We're aware of people encountering a number of download issues today. Anyone still encountering issues should remove all local files, re-clone the repository, and [request a new download link](https://ai.meta.com/resources/models-and-libraries/llama-downloads/). It's critical to do all of these in case you have local corrupt files. When you receive the email, copy *only* the link text - it should begin with https://download.llamameta.net and not with https://l.facebook.com, which will give errors.**
 
+
+### Download
+Once your request is approved, you will receive links to download the tokenizer and model files.
+
+Edit the `download.sh` script with the signed url provided in the email to download the model weights and tokenizer.
+
+The download process can also be automated by setting the respective environmental variables (`PRESIGNED_URL` and `TARGET_FOLDER`).
+
```
Comment (ruanslv) [inline README.md:24]: Most of this feel redundant to me. I suggest instead to just add a example below:

"Then to run the script: `./download.sh`. There are optional ENV variables that can be used for automation.

"Then run the download.sh script, passing the URL provided when prompted to start the download (or using the )".

```suggestion
```

**Review by ruanslv — CHANGES_REQUESTED** [review]
Seems like there's a merge conflict?

File: download.sh [hunk 2]
```diff
@@ -3,10 +3,21 @@
 # Copyright (c) Meta Platforms, Inc. and affiliates.
 # This software may be used and distributed according to the terms of the Llama 2 Community License Agreement.
 
-read -p "Enter the URL from email: " PRESIGNED_URL
-echo ""
```
[the remark this answers is not in the document]
Comment (christopherwoodall) [reply download.sh:7]: Depending on the console size the second `read` can make the terminal look "messy"(116 characters long).

**Comment by ruanslv** [comment]
Doesn't look like merge with latest main branch worked, I still see conflicting files.

If you want some help on this I'd need write access to your fork

**Comment by christopherwoodall** [comment]
Thank you @ruanslv, but I am afraid I no longer have cycles to put toward this commit. I believe #290 may cover the same use case, and with it being newer, take less time to catch up.

**Pull request closed without merging**